test/concurrent/ThreadTest.ooc

```diff
@@ -45,15 +45,17 @@ ThreadTest: class extends Fixture {
 				value set(value get() + 1)
 				Thread yield()
 				Time sleepMilli(1)
+				if (Thread currentThread() isCanceled())
```

Comment:
@emilwestergren Hmm, that works too. Creating a `Thread` with a `Func` would mean an uncancelable thread, and a `Thread` with a `Func (Thread)` would create a cancelable thread.

The limitations are that only that func can check if the thread is canceled. If it calls any other functions, it has to pass the thread instance along in order to allow them to check if the thread is canceled. I haven't checked how pretty this integrates with the threadpool either.